ci/check_d_macro_calls.py

```diff
@@ -0,0 +1,301 @@
+        in addition to another check.
+        """
+        (ret, _) = self._is_cond_part_on_var(var)
+        return ret
+
+    def shorten_cond(self, var):
+        """Remove var from conditional
+
+        If is_cond_part_on_var() returns true then remove the part of
+        the conditional that is dependent on the variable.
+        """
+        (ret, new_code) = self._is_cond_part_on_var(var)
+        assert ret
+        self.code = new_code
+        self._set_text(new_code)
+
+    def remove_brace(self):
+        """Strip the open brace from the end of a conditional"""
+        assert self.conditional_brace
+        self._set_text(self.text.rstrip('{ '))
+        self.conditional_brace = False
+
+    def drop_line(self):
+        """Mark a line as not for inclusion."""
+        self.include = False
+
+    def shift_left(self):
+        """Move the code on a line left one tabstop."""
+        self._set_text(self.text[1:])
+
+    def remove_free_ptr(self):
+        """Replace use of D_FREE_PTR"""
+        if 'D_FREE_PTR' in self.code:
+            self._set_text(self._get_text().replace('D_FREE_PTR', 'D_FREE'))
+
+prev_file = None
+def show_patch(lines):
+
+    # Print a number of lines as a patch to stdout, if any of the lines
+    # have been marked for change.
+
+    global prev_file
+    show = False
+    removed = 0
+    for line in lines:
+        if line.changed:
+            show = True
+        if not line.include:
+            show = True
+            removed += 1
+
+    if not show:
+        return
+
+    # If this code-block is being modified anyway, then change use of
+    # D_FREE_PTR
+    for line in lines:
+        line.remove_free_ptr()
+
+    # Strip out and pre and post-amble, patch does not like it if these
+    # are uneven length, and as patches are not long-lived there is
+    # little benefit in keeping them.
+    start = len(lines)
+    end = 0
+    for idx in range(len(lines)):
+        line = lines[idx]
+        if not (line.changed or not line.include):
+            continue
+        if idx < start:
+            start = idx
+        if idx > end:
+            end = idx
+
+    lines = lines[start:end+1]
+
+    start_line = int(lines[0].lineno)
+
+    if lines[0].filename != prev_file:
+        print('--- a/{}'.format(lines[0].filename))
+        print('+++ a/{}'.format(lines[0].filename))
+        prev_file = lines[0].filename
+
+    print('@@ -{},{} +{},{} @@'.format(start_line,
+                                       len(lines),
+                                       start_line,
+                                       len(lines) - removed))
+    for line in lines:
+        if line.changed:
+            print('+{}'.format(line.new_text))
+            print('-{}'.format(line.text))
+        elif not line.include:
+            print('-{}'.format(line.text))
+        else:
+            print(' {}'.format(line.text))
+
+def check_lines(lines):
+    """Run the checks"""
+
+    for idx in range(len(lines)):
```

Comment:
(pylint-consider-using-enumerate) Consider using enumerate instead of iterating with range and len